### PR TITLE
[JTH][MTDSA-778] Remove unused code.

### PR DESCRIPTION
**Pull request by GlassAndOneHalf**
- Remove unused configuration from application.conf
- Remove all code living under `api` packages, along with associated tests.
- Remove duplicate classes, for example `ErrorCode`, which existed under
both `api` and `models` packages.
- Refactor DeleteExpiredDataService to work with the new resource model.
	- Update lastModifiedDateTime to use `DateTime` in all places.

### CODE REVIEW

**Comment by rakeshavasarala** [comment]
can you also remove the unused configurations from conf/application.conf file like update-tax-year-properties.enabled = true and feature-switch

**Comment by alexec** [comment]
Important to check for accidentally deleted tests in this PR. That often happens by accident with refactoring.

**Comment by alexec** [comment]
Also - failing test on Travis.

File: test/uk/gov/hmrc/selfassessmentapi/repositories/domain/calculations/LossBroughtForwardSpec.scala [hunk 1]
```diff
@@ -1,47 +0,0 @@
...
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package uk.gov.hmrc.selfassessmentapi.repositories.domain.calculations
-
-import uk.gov.hmrc.selfassessmentapi.UnitSpec
-import uk.gov.hmrc.selfassessmentapi.repositories.domain.builders.SelfEmploymentBuilder
-
-class LossBroughtForwardSpec extends UnitSpec {
```
Comment (alexec) [inline test/uk/gov/hmrc/selfassessmentapi/repositories/domain/calculations/LossBroughtForwardSpec.scala]: Deleted by accident?

Comment (GlassAndOneHalf) [inline test/uk/gov/hmrc/selfassessmentapi/repositories/domain/calculations/LossBroughtForwardSpec.scala]: This was a liability calculation test.

**Comment by GlassAndOneHalf** [comment]
The job used to drop mongo collections was only used to control the size of the liabilities collection stored in AWS. As we no longer have this, the job is no longer necessary.

**Pull request merged**